web/api/v1/api.go

```diff
@@ -849,50 +849,80 @@ type recordingRule struct {
 func (api *API) rules(r *http.Request) apiFuncResult {
 	ruleGroups := api.rulesRetriever.RuleGroups()
 	res := &RuleDiscovery{RuleGroups: make([]*RuleGroup, len(ruleGroups))}
+	typeParam := strings.ToLower(r.URL.Query().Get("type"))
+
+	if typeParam != "" && typeParam != "alert" && typeParam != "record" {
+		level.Warn(api.logger).Log("msg", "Invalid type param recieved. Will send all rules.")
```

Comment:
Just send back an API error like https://github.com/prometheus/prometheus/blob/46bbac20126ed43654eb256368b618eaa86b3aa3/web/api/v1/api.go#L314-L318 in this case.

I just noticed that the equivalent code for the targets (https://github.com/prometheus/prometheus/blob/46bbac20126ed43654eb256368b618eaa86b3aa3/web/api/v1/api.go#L612-L613) doesn't fail on malformed input, but it probably should too.